sw/device/lib/dif/dif_rstmgr.c

```diff
@@ -155,6 +164,86 @@ dif_rstmgr_result_t dif_rstmgr_reset_info_clear(const dif_rstmgr_t *handle) {
   return kDifRstmgrOk;
 }
 
+dif_rstmgr_result_t dif_rstmgr_alert_info_set_enabled(
+    const dif_rstmgr_t *handle, dif_rstmgr_toggle_t state) {
+  if (handle == NULL) {
+    return kDifRstmgrBadArg;
+  }
+
+  mmio_region_t base_addr = handle->params.base_addr;
+
+  uint32_t enabled = (state == kDifRstmgrToggleEnabled) ? 0x1 : 0x0;
+
+  // This will clobber the `ALERT_INFO_CTRL.INDEX` field. However, the index
+  // field is only relevant during the crash dump read operation, and is
+  // set by the caller and not the hardware, so it is safe to clobber it.
+  mmio_region_write32(base_addr, RSTMGR_ALERT_INFO_CTRL_REG_OFFSET, enabled);
+
+  return kDifRstmgrOk;
+}
+
+dif_rstmgr_result_t dif_rstmgr_alert_info_get_enabled(
+    const dif_rstmgr_t *handle, dif_rstmgr_toggle_t *state) {
+  if (handle == NULL || state == NULL) {
+    return kDifRstmgrBadArg;
+  }
+
+  mmio_region_t base_addr = handle->params.base_addr;
+
+  uint32_t reg =
+      mmio_region_read32(base_addr, RSTMGR_ALERT_INFO_CTRL_REG_OFFSET);
+  bool enabled = bitfield_bit32_read(reg, RSTMGR_ALERT_INFO_CTRL_EN);
+
+  *state = enabled ? kDifRstmgrToggleEnabled : kDifRstmgrToggleDisabled;
+
+  return kDifRstmgrOk;
+}
+
+dif_rstmgr_result_t dif_rstmgr_alert_info_dump_get_size(
+    const dif_rstmgr_t *handle, size_t *count) {
+  if (handle == NULL || count == NULL) {
+    return kDifRstmgrBadArg;
+  }
+
+  mmio_region_t base_addr = handle->params.base_addr;
+
+  *count = rstmgr_alert_info_dump_get_size(base_addr);
+
+  return kDifRstmgrOk;
+}
```

Comment:
Yes, I don't think this is needed anymore.

I was thinking if a `..._has_dump` would be useful but I don't think it is necessary because clients can infer this by calling `dif_rstmgr_alert_info_dump_read` and checking its size. Up to you.